conda-recipe/meta.yaml

```diff
@@ -25,7 +25,8 @@ requirements:
     - python
     - setuptools
     - numba   {{ NUMBA_VERSION }}
-    - numpy
+    - numpy                           # [win or osx or py==38]
+    - numpy 1.17                      # [linux and py==37]
```

Comment:
This adds constraints on the runtime numpy version too, so generally it's needed to add restrictions in run section as well (see https://numpy.org/devdocs/user/depending_on_numpy.html)

Reply:
I don't see any recommendation to set restrictions in run area if we need to restrict something only for build time. Please point me to right sentence in provided article. More over if we'll set restrictions in run area as well this package will not work in IDP because will need to use numpy 1.17 which was not planned to be a part of IDP 2021.3. This information is a part of Todd's answer in email.

Reply:
No, not == 1.17 in the run section, but conda should know that it has to use compatible version at runrime, see the following sentence: "For conda-forge it’s a little less complicated: there’s dedicated handling for NumPy in build-time and runtime dependencies, so typically this is enough (see here for docs)" and example afterwards.

Reply:
i.e. change:
``` diff
@@ -34,7 +34,7 @@ requirements:
 
   run:
     - python
-    - numpy
+    - {{ pin_compatible('numpy') }}
     - pandas  {{ PANDAS_VERSION }}
     - pyarrow {{ PYARROW_VERSION }}
     - numba   {{ NUMBA_VERSION }}

```

Reply:
@kozlov-alexey , that should work, as I believe building with 1.17.5 will result in a runtime dependency of:
numpy >=1.17.5,<2.0a0